src/screens/gameScreen.ts

```diff
@@ -0,0 +1,48 @@
+import { Entity } from '../entities/entity';
+import { ScreenElement } from './screenElement';
+
+/**
+ * Represents a game screen in the application.
+ * @interface
+ */
+export interface GameScreen {
+  /**
+   * Indicates whether the game screen is currently active.
+   * @type {boolean}
+   */
+  isActive: boolean;
```

Comment:
I think instead of each screen keeping track of whether it's active/displayed, your screen manager should have a field `currentlyActiveScreen: GameScreen`, which keeps track of that for you.

Reply:
There will be circumstances where there will be multiple active screens. ex. The item world screen and the construction screen, players will be able to move around world with the construction menu open to select building entities.

Reply:
You can use an array, in that case. An array might also come in handy if you ever need to layer screens in a particular order. Just a suggestion though.

Reply:
I appreciate the suggestion, but the challenge lies in requiring the ability to access screens by their respective IDs while simultaneously being able to iterate through them.

Reply:
This is outdated. Here is the current implementation.
```
update(): void {
    const { keysPressed } = this.inputService;
    this.updateScreenElements();
    this.gameScreens.activeScreens.getScreens().forEach((screen) => {
      screen.handleInput(
        keysPressed,
        this.gameScreens,
        this.screenKeyController
      );
    });
  }